src/OpenSage.Game/Logic/Orders/OrderProcessor.cs

```diff
@@ -106,12 +106,12 @@ public void Process(IEnumerable<Order> orders)
                             foreach (var unit in player.SelectedUnits)
                             {
                                 // Only units that can produce stuff should produce it
-                                var productionJob = unit?.ProductionUpdate.ProductionQueue[queueIndex];
+                                var productionJob = unit.ProductionUpdate?.ProductionQueue[queueIndex];
                                 var objectDefinition = productionJob.ObjectDefinition;
```

Comment:
This still has the problem, that if the line above results in a `null` object, then this line will crash.